admin/services/data-access/check-window.data.service.js

```diff
@@ -49,47 +94,151 @@ const checkWindowDataService = {
       .find(query)
       .sort(sorting)
       .exec()
+  },
+    /**
+   * Fetch check windows by status, sort by, sort direction and date (current or past).
+   * @param isDeleted
+   * @param sortBy valid values are [checkWindowName|adminStartDate|checkStartDate]
+   * @param sortDirection valid values are [asc|desc]
+   * @param isCurrent
+   * @returns {Promise.<void>}
+   */
+  sqlFetchCheckWindows: async (sortBy, sortDirection, isDeleted, isCurrent) => {
+    const currentTimestamp = moment.utc().toDate()
+    let criteria = isDeleted ? 'isDeleted=1' : 'isDeleted=0'
+
+    if (isCurrent === true) {
+      isDeleted += ` AND checkEndDate >= @currentTimestamp`
+    } else {
+      isDeleted += ` AND checkEndDate <= @currentTimestamp`
+    }
+    sortDirection = sortDirection !== 'asc' ? 'desc' : 'asc'
+    switch (sortBy) {
+      case 'checkWindowName':
+        sortBy = 'name'
+        break
+      case 'adminStartDate':
+      case 'checkStartDate':
+      // all 3 are acceptable as-is
+        break
+      default:
+      // anything else should default to checkWindow name
+        sortBy = 'name'
+    }
+
+    const sql = `SELECT * FROM [mtc_admin].[vewCheckWindowsWithFormCount] WHERE ${criteria} ORDER BY ${sortBy} ${sortDirection}`
+    const params = [
+      {
+        name: 'currentTimestamp',
+        value: currentTimestamp,
+        type: TYPES.DateTimeOffset
+      }
+    ]
+    return sqlService.query(sql, params)
   },
   /**
    * Fetch (one) check window for present date.
+   * @deprecated use sqlFetchCurrentCheckWindow
    * @returns {Promise.<*>}
    */
   fetchCurrentCheckWindow: async () => {
+    winston.warn('deprecated. use check-window.data.service.sqlFetchCurrentCheckWindow')
     const now = new Date()
     const checkWindow = await CheckWindow.findOne({startDate: {$lte: now}, endDate: {$gte: now}}).exec()
     if (!checkWindow) {
       throw new Error('No check-window is currently available')
     }
     return checkWindow
+  },
+    /**
+   * Fetch (one) check window for present date.
+   * @returns {Promise.<*>}
+   */
+  sqlFetchCurrentCheckWindow: async () => {
+    const now = moment.utc()
+    const sql = `SELECT * FROM [mtc_admin].[checkWindow] WHERE 
+      adminStartDate <=@currentTimestamp AND
+       adminStartDate >=@currentTimestamp`
+    const params = [
+      {
+        name: 'currentTimestamp',
+        value: now,
+        type: TYPES.DateTimeOffset
+      }
+    ]
+    return sqlService.query(sql, params)
   },
   /**
    * Fetch (non-deleted) current check windows by sort by, sort direction
    * @param sortBy
    * @param sortDirection
    * @returns {Promise.<*|Promise.<void>>}
    */
-  fetchCurrentCheckWindows: async (sortBy, sortDirection) => {
-    return checkWindowDataService.fetchCheckWindows(sortBy, sortDirection, false, true)
+  sqlFetchCurrentCheckWindows: async (sortBy, sortDirection) => {
```

Comment:
Whilst I am not sure why we need this method, it might be better names `sqlFindCurrentCheckWindows()`